docs/source/tutorial/10-authenticate-your-queries.mdx

```diff
@@ -0,0 +1,180 @@
+---
+title: "10. Authenticate your queries"
+---
+
+In this section, you will book a flight 🚀! Booking a flight requires being authenticated to the server so the correct person is sent to space! To do that, and since Apollo Android is using [OkHttp](https://square.github.io/okhttp/) to handle HTTP requests, you will leverage the OkHttp [Interceptors](https://square.github.io/okhttp/interceptors/) to add headers to your GraphQL requests.
+
+## Add the interceptor
+
+In `Apollo.kt`, add the `AuthorizationInterceptor` class:
+
+```kotlin:title=app/src/main/kotlin/com/example/rocketreserver/Apollo.kt
+private class AuthorizationInterceptor(val context: Context): Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request().newBuilder()
+            .addHeader("Authorization", User.getToken(context) ?: "")
+            .build()
+
+        return chain.proceed(request)
+    }
+}
+```
+
+This interceptor will append a `"Authorization: $token"` HTTP header to every request.
+
+## Use the interceptor
+
+Create a custom OkHttpClient that will use this interceptor and pass it to the ApolloClient:
+
+```kotlin:title=app/src/main/kotlin/com/example/rocketreserver/Apollo.kt
+    instance = ApolloClient.builder()
+        .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/")
+        .okHttpClient(OkHttpClient.Builder()
+            .addInterceptor(AuthorizationInterceptor(context))
+            .build()
+        )
+        .build()
+```
+
+Notice that `Apollo.kt` now requires a context to create the interceptor and read the token from the `EncryptedSharedPreferences`. `apolloClient` cannot be a top level variable anymore so transform it to a top-level method that takes a `context` parameter instead:
+
+```kotlin:title=app/src/main/kotlin/com/example/rocketreserver/Apollo.kt
+fun apolloClient(context: Context): ApolloClient {
+    return ApolloClient.builder()
+        .serverUrl("https://apollo-fullstack-tutorial.herokuapp.com/")
+        .okHttpClient(OkHttpClient.Builder()
+            .addInterceptor(AuthorizationInterceptor(context))
+            .build()
+        )
+        .build()
+}
+```
+
+Replace all usages of `apolloClient` with `apolloClient(context)`. There is one instance in each fragment:
+
+```kotlin
+// LaunchListFragment
+apolloClient(requireContext()).query(LaunchListQuery(cursor = Input.fromNullable(cursor))).toDeferred().await()
+
+// LaunchDetailsFragment
+apolloClient(requireContext()).query(LaunchDetailsQuery(id = args.launchId)).toDeferred().await()
+
+// LoginFragment
+apolloClient(requireContext()).mutate(LoginMutation(email = Input.fromNullable(email))).toDeferred().await()
+
+```
+
+Make the `apolloClient` a singleton
+
+There is no need to create a new instance of the apolloClient for each GraphQL request. You can reuse the `apolloClient`:
```

Comment:
Would it be ok to mention that users are better to use a dependency injection to manage the singleton `apolloClient` instance and make it clear that this is done like this only for sample purposes.

Reply:
This is a part I wasn't sure how to explain best. I'd like to avoid adding Dagger or any other DI framework to the tutorial but this is better to explicit that this is for sample use only.

I added that note, let me know how it sounds:
```
**Note**: Here the singleton handled manually for sample purposes. In a real life application, you would certainly use a dependency injection framework such as `Dagger` or `Koin`.
```

Reply:
For sure. This sounds good.